pilot/pkg/security/authn/v1beta1/policy_applier.go

```diff
@@ -216,6 +218,14 @@ func NewPolicyApplier(rootNamespace string,
 	}
 }
 
+// Remove all Non-Alphanumeric Characters from a jwksURI and append to "n"
+func createFakeJwks(jwksURI string) string {
+	encodedString := b64.URLEncoding.EncodeToString([]byte(jwksURI))
+	reg := regexp.MustCompile("[^a-zA-Z0-9]+")
```

Comment:
No need to replace if you're already using base64.

Reply:
But I noticed base64 encoded string will still contains special character https://gobyexample.com/base64-encoding. it's safe to guarantee jwks n filed with only Alphanumeric Characters.

Reply:
`-` and `_` is allowed in `n` but not sure if it's valid if an `n` ends with dash.

Reply:
Can you test if it is allowed to be end of `n`? I'm surprised if it doesn't support this...

And if you're doing this replacement, why would you need base64 anyway?

Reply:
@xulingqing this is still not updated, have you tested it?

Reply:
I've tested it and the n field can end with "-". I'm surperised why it didn't get  updated..